server/zoom/oauth.go

```diff
@@ -95,15 +99,42 @@ func (c *OAuthClient) GetMeeting(meetingID int) (*Meeting, error) {
 }
 
 func (c *OAuthClient) getUserViaOAuth(user *model.User) (*User, error) {
-	client := c.config.Client(context.Background(), c.token)
 	url := fmt.Sprintf("%s/users/me", c.apiURL)
+	currentToken := c.token
 	if c.isAccountLevel {
 		url = fmt.Sprintf("%s/users/%s", c.apiURL, user.Email)
+		savedToken, err := c.api.KVGet(zoomSuperUserTokenKey)
+		if err != nil {
+			log.Printf("error getting user token: %s", err)
+		}
+
+		uErr := json.Unmarshal(savedToken, &currentToken)
+		if uErr != nil {
+			log.Printf("error parsing user token: %s", err)
+		}
+
+		tokenSource := c.config.TokenSource(context.Background(), currentToken)
+		updatedToken, tsErr := tokenSource.Token()
+
+		if tsErr == nil {
+			if updatedToken.AccessToken != currentToken.AccessToken {
+				newToken, _ := json.Marshal(updatedToken)
```

Comment:
It's not obvious what the variables `updatedToken` and `newToken` are holding. `tokenBytes` may be a better name here.

```suggestion
				tokenBytes, _ := json.Marshal(updatedToken)
```